parts/k8s/addons/1.10/kubernetesmasteraddons-kube-dns-deployment.yaml

```diff
@@ -52,7 +52,7 @@ metadata:
   name: kube-dns
   namespace: kube-system
   labels:
-    addonmanager.kubernetes.io/mode: EnsureExists
+    addonmanager.kubernetes.io/mode: Reconcile
```

Comment:
I think we probably want to keep this as `EnsureExists` as a `Reconcile` might blow away a pre-existing `ConfigMap` with data in it.

Ref:

https://github.com/kubernetes/kubernetes/blob/master/cluster/addons/dns/kube-dns/kube-dns.yaml.in

Reply:
updated all ConfigMaps to use `EnsureExists`

Reply:
I'm torn about the omsagent ConfigMap since it data that is part of the ConfigMap resource definition, wondering I'm thinking we want to leave it to `Reconcile` so changes to that data get picked up by updates (it's currently set to `Reconcile`): 

https://github.com/Azure/aks-engine/blob/403a11c1ea84c0fbadeb07e81b1a7c97d500fd59/parts/k8s/containeraddons/kubernetesmasteraddons-omsagent-daemonset.yaml#L51